rustfmt.toml

```diff
@@ -0,0 +1,29 @@
+tab_spaces = 4
+newline_style = "Auto"
+indent_style = "Block"
+wrap_comments = true
+format_code_in_doc_comments = false
+group_imports = "StdExternalCrate"
+reorder_imports = true
+reorder_modules = true
+reorder_impl_items = false
+type_punctuation_density = "Wide"
+space_before_colon = false
+space_after_colon = true
+spaces_around_ranges = true
+binop_separator = "Front"
+remove_nested_parens = true
+combine_control_expr = true
+force_multiline_blocks = false
+fn_params_layout = "Tall"
+brace_style = "SameLineWhere"
+control_brace_style = "AlwaysSameLine"
+trailing_semicolon = true
+trailing_comma = "Vertical"
+match_block_trailing_comma = false
+merge_derives = true
+use_try_shorthand = true
+use_field_init_shorthand = true
+force_explicit_abi = true
+condense_wildcard_suffixes = true
+edition = "2021"
```

Comment:
Why set so many settings?

Reply:
Most of them are default but incase somebody has a local user wide fmt config this will override that

Reply:
Maybe add a comment for which ones aren't the default and why you reset the defaults

Reply:
Actually, I didn't realize how many of those are unstable because I'm always on nightly, our ci can't utilize the nightly ones so ill trim the toml down alot